### PR TITLE
Add (enhanced) ParsingSeparator to base library

### PR DESCRIPTION
**Pull request by thpr**

### CODE REVIEW

**Comment by LegacyKing** [comment]
No travis checks on this, and jenkins won't run till this is merged. Are you expecting peer review?

**Comment by thpr** [comment]
Had dropped a note to Connor to see if he wanted to review it

File: PCGen-base/code/src/java/pcgen/base/text/ParsingSeparator.java [hunk 1]
```diff
@@ -0,0 +1,304 @@
...
+	 * @param end
+	 *            The ending character for a grouping pair
+	 * @return true if the grouping pair was added to this ParsingSeparator
+	 */
+	public boolean addGroupingPair(char start, char end)
+	{
+		//Can't change after we've started parsing
+		if (started)
+		{
+			return false;
```
Comment (cpmeister) [inline PCGen-base/code/src/java/pcgen/base/text/ParsingSeparator.java]: why not throw an IllegalStateException at this point? I would prefer funny business to be highly visible.

**Comment by thpr** [comment]
Fixed, but could use another review just in case

**Comment by cpmeister** [comment]
My only remaining nitpick:
Stack is almost a deprecated collection. Although the use of Stack gives a clear intent on how your algorithm should work it is a synchronized collection that you are using in only a single thread. For performance could you use a Deque implementation like LinkedList or ArrayDeque instead?

**Pull request merged**